lib/erb_lint/linters/hard_coded_string.rb

```diff
@@ -26,15 +26,23 @@ def run(processed_source)
         hardcoded_strings = processed_source.ast.descendants(:text).each_with_object([]) do |text_node, to_check|
           next if javascript?(processed_source, text_node)
 
-          offended_strings = text_node.to_a.select { |node| relevant_node(node) }
-          offended_strings.each do |offended_string|
-            offended_string.split("\n").each do |str|
-              to_check << [text_node, str] if check_string?(str)
+          text_node.to_a.each do |node|
+            unless string = relevant_node(node)
+              to_check << nil
+              next
+            end
+
+            if node.is_a?(String)
+              string.split("\n").each do |str|
+                append_to_check(to_check, str, text_node)
+              end
+            elsif node&.type == :erb
+              append_to_check(to_check, string, text_node)
             end
           end
         end
 
-        hardcoded_strings.compact.each do |text_node, offended_str|
+        clear_solo_erb_code(hardcoded_strings).each do |text_node, offended_str|
```

Comment:
Since we can add the source of an `erb_node` to `to_check` (because we have no way to determine if the next node will be a string and therefore the `erb_node` would be the first part of the string ex (`<%= ui_link("See more", path_see_more) %> of our awesome strings`)
So we need to remove the nodes that are not part of an interpolation